### PR TITLE
Feature/843 get instance delegation

### PR DESCRIPTION
**Pull request by lovoll**
## Related Issue(s)
- #{insert issue number here}

## Developer/Reviewer Checklist
- [ ] **Your** code builds clean without any errors or warnings
- [ ] No changes to config/appsettings or environment variables created in separate linked PR
- [ ] Manual testing done (required)
- [ ] Relevant integration test added (required for functional changes)
- [ ] Relevant automated test added (required for functional changes)
- [ ] All integration tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.AccessManagement.Core/Enums/InstanceDelegationSource.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        /// <summary>
+        /// Normal instance delegation
+        /// </summary>
+        [EnumMember(Value = "user")]
+        [PgName("user")]
+        User,
+
+        /// <summary>
+        /// Special case of instance delegation extends delegations to organizations to all users in the receiving organization with parallel role/package to also getting access to the instance
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement.Core/Enums/InstanceDelegationSource.cs:19]: Copy/paste summary fra InstanceDelegationMode enum?

File: src/Altinn.AccessManagement.Core/Services/AppsInstanceDelegationService.cs [hunk 2]
```diff
@@ -359,14 +365,70 @@ private async Task<List<InstanceRightDelegationResult>> DelegateRights(InstanceR
...
+            try
+            {
+                delegableRights = await _pip.GetDelegableRightsByApp(rightsQueryForApp, cancellationToken);
+            }
+            catch (ValidationException)
+            {
+                errors.Add(ValidationErrors.MissingPolicy, "request.Resource");
+            }
+
+            if (delegableRights == null || !delegableRights.Exists(r => r.CanDelegate.HasValue && r.CanDelegate.Value))
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement.Core/Services/AppsInstanceDelegationService.cs:399]: Burde kanskje dokumentert/kommentert på at dette i praksis er Autorisasjonssjekken på at autentisert app har minst _en_ tilgang for å i det hele tatt få lov til å gjøre get oppslaget

File: src/Altinn.AccessManagement.Core/Enums/InstanceDelegationSource.cs (hunk 1 above)
Comment (lovoll) [reply src/Altinn.AccessManagement.Core/Enums/InstanceDelegationSource.cs]: Fixed

File: src/Altinn.AccessManagement.Persistence/DelegationMetadataRepo.cs [hunk 3]
```diff
@@ -460,6 +462,66 @@ INSERT INTO delegation.instancedelegationchanges(
...
+	            ,touuid
+	            ,totype
+	            ,performedby
+	            ,performedbytype
+	            ,blobstoragepolicypath
+	            ,blobstorageversionid
+	            ,created
+            FROM
+	            LatestChanges lc
+	            JOIN delegation.instancedelegationchanges dc ON lc.instancedelegationchangeid = dc.instancedelegationchangeid;";
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement.Persistence/DelegationMetadataRepo.cs:505]: Burde kanskje filtrere bort RevokeLast delegeringer her i databaselaget i stedet for å stole på at service implementasjoner husker det

**Review by jonkjetiloye — APPROVED** [review]

**Pull request merged**